### PR TITLE
OSDOCS 6914: Updating pod examples to comply with restricted PSA (App…

### PR DESCRIPTION
**Pull request by bergerhoffer**
…lications book)

Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-6914

Link to docs preview:
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-examples
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-configuring_application-health
* https://62470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/applications/quotas/quotas-setting-per-project.html#setting-resource-quota-for-extended-resources_quotas-setting-per-project

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@s-urbaniak @stlaz PTAL

File: modules/application-health-about.adoc [hunk 1]
```diff
@@ -104,6 +112,10 @@ metadata:
   labels:
     test: health-check
   name: my-application
+  securityContext:
```
Comment (s-urbaniak) [inline modules/application-health-about.adoc:115]: oh sec, this needs to move under `spec`

**Comment by s-urbaniak** [comment]
until https://github.com/openshift/openshift-docs/pull/62470#discussion_r1266819211 is resolved

**Comment by bergerhoffer** [comment]
FYI @Preeticp of this PR to update all pod examples in the applications book to comply with pod security admission standards. This is part of a larger effort to update all pod examples in the repo. Let me know if you have any questions or concerns on this. Thanks!

**Comment by bergerhoffer** [comment]
@xingxingxia @gkarager This PR is ready for QE review. Again, please let me know if we should request QEs from this area to do the verification.

Preview links of the affected files are in the description.

**Comment by xingxingxia** [comment]
@bergerhoffer , @gkarager is PTO, I saw your Jira issue's tagging. I was busy in Auth / cert-manager two parts work. So kindly expect one or two days' waiting before I check, thanks! Will check later.

**Comment by bergerhoffer** [comment]
Thanks @xingxingxia! No hurry at all, I'm just letting you know which PRs are actually ready for QE review when you all get a chance. Some of the other PRs in the Jira still have open questions in them that we need to resolve first. Thanks!

**Comment by xingxingxia** [comment]
> Again, please let me know if we should request QEs from this area to do the verification.

@bergerhoffer from the perspective of adding the "securityContext" fields, it is LGTM. From the verification of user-end, yes, need related QEs to help verify. At least I tried one of the modules/application-health-about.adoc, it has issue:
```
[xxia@2023-07-20 17:34:05 CST my]$ oc create -f <(echo "
apiVersion: v1
kind: Pod
metadata:
  labels:
    test: health-check
  name: my-application
spec:
  securityContext:
    runAsNonRoot: true
    seccompProfile:
      type: RuntimeDefault
  containers:
  - name: goproxy-app
    args:
    image: registry.k8s.io/goproxy:0.1
    readinessProbe:
      exec:
        command:
        - cat
        - /tmp/healthy
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop: [ALL]
")
pod/my-application created

# "0/1" shows the container is not ready:
[xxia@2023-07-20 17:34:21 CST my]$ oc get po
NAME             READY   STATUS    RESTARTS   AGE
my-application   0/1     Running   0          2m19s

[xxia@2023-07-20 17:36:32 CST my]$ oc describe po my-application
Name:             my-application
Namespace:        test
...
Events:
  Type     Reason          Age    From               Message
  ----     ------          ----   ----               -------
...
  Warning  Unhealthy  4s (x8 over 74s)  kubelet  (combined from similar events): Readiness probe errored: rpc error: code = Unknown desc = command error: time="2023-07-20T09:36:43Z" level=error msg="exec failed: unable to start container process: exec: \"cat\": executable file not found in $PATH"
, stdout: , stderr: , exit code -1
```
@gkarager comes back :smile: today, please help handle new tagging when there are, thanks : )

**Comment by bergerhoffer** [comment]
Hi @lyman9966, I'm looking for someone to help test/verify these updates.  This PR updates all pod examples in the applications book to comply with pod security admission standards. This is part of a larger effort to update all pod examples in the repo. It looks like at least one of them has an issue when being tested (see comment above). 

I'm not sure which area owns these docs, but I found your name in verifying an old PR in this area. But let me know if someone else should help with this instead. Thanks!

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.15` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.14. And any PR going into `main` must also target the latest version branch (enterprise-4.15).

If the update in your PR does NOT apply to version 4.15 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main

**Comment by lyman9966** [comment]
@bergerhoffer I can't open the docs preview, it showed: Site Not Found (e.g. https://62470--docspreview.netlify.app/openshift-enterprise/latest/applications/application-health.html#application-health-examples)

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.16` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.15. And any PR going into `main` must also target the latest version branch (enterprise-4.16).

If the update in your PR does NOT apply to version 4.16 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by bergerhoffer** [comment]
HI @lyman9966, sorry for the long delay. I've updated the preview links (we changed preview systems). All should work now. Let me know how these changes look, thanks!

File: modules/application-health-about.adoc [hunk 2]
```diff
@@ -86,6 +90,10 @@ spec:
         command: <5>
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply modules/application-health-about.adoc:90]: @lyman9966 Do you happen to know a different image we could use here instead?

[the remark this answers is not in the document]
Comment (lyman9966) [reply modules/application-health-about.adoc:90]: Sorry, I don't have this kind of image on hand. And I think we can't simply replace it with any image which satisfied PSA, but it should also support the command of readinessProbe:
readinessProbe:
      exec:
        command:
        - cat
        - /tmp/healthy

**Pull request closed without merging**